From 7fd150734c4febeeda0ac680211e220fbebf97de Mon Sep 17 00:00:00 2001
From: Jesus Camacho Rodriguez <jcamacho@apache.org>
Date: Fri, 6 Nov 2015 17:27:56 +0100
Subject: [PATCH 0591/1164] HIVE-12344: Wrong types inferred for SemiJoin
 generation in CBO (Jesus Camacho Rodriguez,
 reviewed by Laljo John Pullokkaran)

Change-Id: If4fc918f6b4c626a79efc35265bc8877850865db
---
 .../hive/ql/optimizer/calcite/HiveCalciteUtil.java |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/HiveCalciteUtil.java b/ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/HiveCalciteUtil.java
index 298855a..8e3594d 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/HiveCalciteUtil.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/HiveCalciteUtil.java
@@ -213,8 +213,8 @@ public static RexNode projectNonColumnEquiConditions(ProjectFactory factory, Rel
       leftKeys.add(origLeftInputSize + i);
       rightKeys.add(origRightInputSize + i);
       RexNode cond = rexBuilder.makeCall(SqlStdOperatorTable.EQUALS,
-          rexBuilder.makeInputRef(newLeftFields.get(i).getType(), newLeftOffset + i),
-          rexBuilder.makeInputRef(newLeftFields.get(i).getType(), newRightOffset + i));
+          rexBuilder.makeInputRef(newLeftFields.get(origLeftInputSize + i).getType(), newLeftOffset + i),
+          rexBuilder.makeInputRef(newRightFields.get(origRightInputSize + i).getType(), newRightOffset + i));
       if (outJoinCond == null) {
         outJoinCond = cond;
       } else {
-- 
1.7.9.5

